exporter/sumologicexporter/filter.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type filter struct {
+	regexes []*regexp.Regexp
+}
+
+// Fields represents concatenated metadata
+type Fields string
+
+func newFilter(fields []string) (filter, error) {
+	metadataRegexes := make([]*regexp.Regexp, len(fields))
+
+	for i, field := range fields {
+		regex, err := regexp.Compile(field)
+		if err != nil {
+			return filter{}, err
+		}
+
+		metadataRegexes[i] = regex
+	}
+
+	return filter{
+		regexes: metadataRegexes,
+	}, nil
+}
+
+// convertAttributeToString returns string which represents given pdata.AttributeValue
+func (f *filter) convertAttributeToString(value pdata.AttributeValue) string {
```

Comment:
This function looks similar to https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/protospan_translation.go#L114, you may want to reuse that

Reply:
Definitely this is what I need. Thank you!

Reply:
I can read your message in two ways. 1. What is available in core is not useful, what is implemented here is what I need. 2. What is in core is what I need, I will use that.

Which of the versions are you referring to?

Reply:
I'm referring to the second version. Sorry for confusion 😃

Reply:
Done, please look again